src/query/functions.rs

```diff
@@ -282,6 +304,20 @@ mod validate {
         }
     }
 
+    pub fn arg_type_list_of_list_of_strings (arg: &DataType) -> Result<Vec<Vec<String>>, QueryError> {
+        let mut tagged_lists = arg_type_list(arg)?.clone();
+        let mut lists: Vec<Vec<String>> = Vec::new();
+        for list in tagged_lists.drain(..) {
+            match list {
+                DataType::List(_) => lists.push(arg_type_string_list(&list)?.clone()),
+                ref invalid_type => return Err(QueryError::InvalidFunctionParameters(
+                    format!("Expected function parameter of type list of tuples of strings, list contains {:?}", invalid_type)
```

Comment:
You don't check for a tuple here, you check for an array. An array of a forced length would be similar to a tuple though as long as the tuple contains only one type if that's what you want to achieve.